src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -39,184 +39,236 @@
 
 #include <OpenMS/CONCEPT/LogStream.h>
 
-#include <seqan/basic.h>
-#include <seqan/stream.h>
-#include <seqan/seq_io/guess_stream_format.h>
-#include <seqan/seq_io/read_fasta_fastq.h>
 
 namespace OpenMS
 {
-  using namespace std;
-  typedef seqan::RecordReader<std::fstream, seqan::SinglePass<> > FASTARecordReader;
-
-  FASTAFile::FASTAFile()
-  : reader_(std::nullptr_t()), // point to nothing
-    entries_read_(0)
-  {
-  }
-  
-  FASTAFile::~FASTAFile()
-  {
-    // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
-  }
-
-  void FASTAFile::readStart(const String& filename)
-  {
-    if (!File::exists(filename))
+    using namespace std;
+
+    FASTAFile::FASTAFile()
+            : entries_read_(0)
     {
-      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
     }
 
-    if (!File::readable(filename))
+    FASTAFile::~FASTAFile()
     {
-      throw Exception::FileNotReadable(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
     }
 
-    if (infile_.is_open()) infile_.close(); // precaution
+    bool FASTAFile::readEntry_(std::string & id, std::string & seq)
+    {
+        std::streambuf* sb = infile_.rdbuf();
+        bool condition = true;
 
-    infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
+        while(sb->sgetc() == '#')// Skip the header of PEFF files (http://www.psidev.info/peff)
+        {
+            infile_.ignore(numeric_limits<streamsize>::max(),'\n');
+        }
+        if(sb->sbumpc() == '>')//not saving '>'
+        {
+            while(condition)// reading the ID
+            {
+                int c = sb->sbumpc();// get and advance to next char
+                switch (c)
+                {
+                    case '\n'://ID finished
+                        condition = false;
+                        break;
+                    case std::streambuf::traits_type::eof():
+                        infile_.setstate(std::ios::eofbit);
+                        if (id.empty())
```

Comment:
should you not return false in any case here? You did not read a sequence yet..

Add a test for this.